skimage/filters/_frangi.py

```diff
@@ -0,0 +1,176 @@
+import numpy as np
+
+__all__ = ['frangi', 'hessian']
+
+
+def _frangi_hessian_common_filter(image, scale, scale_step, beta1, beta2):
+    """This is an intermediate function for Frangi and Hessian filters.
+
+    Shares the common code for Frangi and Hessian functions.
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Array with input image data.
+    scale : tuple of floats, optional
+        The range of sigmas used.
+    scale_step : float, optional
+        Step size between sigmas.
+    beta1 : float, optional
+        Frangi correction constant.
+    beta2 : float, optional
+        Frangi correction constant.
+
+    Returns
+    -------
+    filtered_list : list
+        List of pre-filtered images.
+
+    """
+    # Import has to be here due to circular import error
+    from ..feature import hessian_matrix, hessian_matrix_eigvals
+
+    sigmas = np.arange(scale[0], scale[1], scale_step)
```

Comment:
@jni I agree. Should have really separate the cases (`<` and `==`).
But I think that the error with `scale==(x, x)` will be quite common. As for me, the expected behaviour here is to have the filtere image with `sigma=x`, but an empty array is created instead.